subsys/fs/fs.c

```diff
@@ -130,7 +130,7 @@ static int fs_get_mnt_point(struct fs_mount_t **mnt_pntp,
 }
 
 /* File operations */
-int fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags)
+static int _fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags)
```

Comment:
Function names should avoid starting with underscores IMO. Maybe `fs_open_locked` shows the intent.

Also maybe add `ALWAYS_INLINE` to avoid the additional stack frame.